langlib/langlib-test/src/test/resources/test-src/tablelib_test.bal

```diff
@@ -683,3 +683,37 @@ function testPutValidDataToKeylessTbl() returns boolean {
     testPassed = testPassed && tableToList[2] == intern1;
     return testPassed;
 }
+
+function testReadOnlyTableFilter() {
+    PersonalTable & readonly personTable = table [
+      { name: "Harry", age: 14 },
+      { name: "Hermione", age: 28 },
+      { name: "Ron", age: 11 },
+      { name: "Draco", age: 23 }
+    ];
+    table<Person> children = personTable.filter(function (Person person) returns boolean {
+                                                      return person.age < 18;
+                                                  });
+    assertEquals(children.length(), 2);
+    children.forEach(function(Person person) {
+        assertTrue(person.age < 18);
```

Comment:
Shall we also assert that each `person` value is immutable?

Reply:
Updated in last commit